infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -167,16 +160,13 @@ public void trim(LogAddress logAddress) {
                 handle.getPendingTrims().contains(logAddress.getAddress())) {
             return;
         }
-
         TrimEntry entry = TrimEntry.newBuilder()
                 .setChecksum(getChecksum(logAddress.getAddress()))
                 .setAddress(logAddress.getAddress())
                 .build();
 
         // TODO(Maithem) possibly move this to SegmentHandle. Do we need to close and flush?
-        OutputStream outputStream = Channels.newOutputStream(handle.getPendingTrimChannel());
```

Comment:
@Maithem I believe this change is correct, but it causes your StreamLogFiles.trim() test to fail on the place where you check the half segment is still not trimmed. Let me see does your test work due to side effect from non-closed stream?